internal/handlers/common/update.go

```diff
@@ -16,35 +16,48 @@ package common
 
 import (
 	"fmt"
+	"sort"
 
 	"github.com/FerretDB/FerretDB/internal/types"
-	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
 // UpdateDocument updates the given document with a series of update operators.
-func UpdateDocument(doc, update *types.Document) error {
+// Returns true if document was changed.
+func UpdateDocument(doc, update *types.Document) (bool, error) {
 	for _, updateOp := range update.Keys() {
 		updateV := must.NotFail(update.Get(updateOp))
 
 		switch updateOp {
 		case "$set":
-			setDoc, err := AssertType[*types.Document](updateV)
-			if err != nil {
-				return err
-			}
 
-			for _, setKey := range setDoc.Keys() {
-				setValue := must.NotFail(setDoc.Get(setKey))
-				if err = doc.Set(setKey, setValue); err != nil {
-					return lazyerrors.Error(err)
+			switch setDoc := updateV.(type) {
```

Comment:
Does it make sense to move this switch to a separate function and call if from here? I guess, the idea is to support more `updateOp`s in the future, and then the parent switch-case will be really huge. Also, if it's a separate function it's possible (and easier) to write tests to cover this specific logic.

Reply:
> Does it make sense to move this switch to a separate function and call if from here?

I think it's small enough, trying to keep it simple. 

> I guess, the idea is to support more `updateOp`s in the future, and then the parent switch-case will be really huge. 

when it becomes then it can be extracted. 

> Also, if it's a separate function it's possible (and easier) to write tests to cover this specific logic.

make sense. But it's covered in integration tests. WDYT of it?